torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -463,7 +462,10 @@ def apply_cp(model, world_mesh, parallel_dims, job_config: JobConfig):
     """
     if parallel_dims.tp_enabled or parallel_dims.pp_enabled:
         raise NotImplementedError("CP + TP or CP + PP are not supported yet.")
-    cp_mesh = world_mesh["cp"]
+    dp_mesh = world_mesh["dp"]
+    cp_mesh = dp_mesh.reshape(
+        (dp_mesh.size() // parallel_dims.cp, parallel_dims.cp), ("dp", "cp")
+    )
```

Comment:
Maybe I am confused about the name again. lol
Should this be 
```
    cp_mesh = dp_mesh.reshape(
        (dp_mesh.size() // parallel_dims.cp, parallel_dims.cp), ("dp", "cp")
    )["cp"]
```?

Or for here we need the 2D mesh, while inside cp we will get the slice for cp_mesh["cp"].

Reply:
No, you are right, it is easy to get messed up with the naming confusion, lol.